frontend/src/metabase-lib/expressions/index.ts

```diff
@@ -52,44 +59,47 @@ export function formatIdentifier(name, { quotes = EDITOR_QUOTES } = {}) {
   return quoteString(name, quotes.identifierQuoteDefault);
 }
 
-function isReservedWord(word) {
+function isReservedWord(word: string) {
   return !!getMBQLName(word);
 }
 
 // METRICS
 
-export function parseMetric(metricName, { legacyQuery }) {
-  return legacyQuery
-    .table()
-    .metrics.find(
-      metric => metric.name.toLowerCase() === metricName.toLowerCase(),
-    );
+export function parseMetric(
+  metricName: string,
+  { legacyQuery }: { legacyQuery: StructuredQuery },
+) {
+  return checkNotNull(legacyQuery.table()).metrics?.find(
+    metric => metric.name.toLowerCase() === metricName.toLowerCase(),
+  );
 }
 
-export function formatMetricName(metric, options) {
+export function formatMetricName(metric: Metric, options: object) {
```

Comment:
Can you please add typing for `options` here and in `formatSegmentName` + `formatDimensionName`?

Reply:
those options are not defined, I changed them to `Record<string, any>`, but I don't like it better